eclipse-collections/src/main/java/org/eclipse/collections/impl/list/immutable/AbstractImmutableList.java

```diff
@@ -666,6 +667,24 @@ public T remove(int index)
         throw new UnsupportedOperationException("Cannot call remove() on " + this.getClass().getSimpleName());
     }
 
+    /**
+     * @since 10.0
+     */
+    @Override
+    public void replaceAll(UnaryOperator<T> operator)
+    {
+        throw new UnsupportedOperationException("Cannot call replaceAll() on " + this.getClass().getSimpleName());
+    }
+
+    /**
+     * @since 10.0
+     */
+    @Override
+    public void sort(Comparator<? super T> c)
+    {
+        throw new UnsupportedOperationException("Cannot call sort() on " + this.getClass().getSimpleName());
+    }
+
```

Comment:
Where is sortThis()? I was expecting it would already exist in this class. I'm sure it always threw UnsupportedOperationException, but was it throwing an unhelpful message like "Cannot call set()" instead of "Cannot call sortThis()"?

Reply:
Do our tests check the message, or just the exception type?

Reply:
I have to admit to not understanding this question at first.  The method `sortThis()` does not exist in `AbstractImmutableList` because it does not extend `MutableList`.  As you know, our `ImmutableList` implementations have to implement `List` and `sort` is a mutating method on `List` so it should throw `UnsupportedOperationException`.  It was previously throwing an unhelpful message as you point out which is why I provided an override.

Reply:
The tests just check the exception type, not the message.

Reply:
Right, I forgot that the interfaces are different here.